lib/async/enumerable.rb

```diff
@@ -41,14 +53,22 @@ def self.included(base)
       base.include(Methods)      # Include all method groups
       base.include(FiberLimiter) # Include fiber limiting functionality
       base.include(AsyncMethod)  # Include async method last to override Enumerable's
+      base.include(ConfigAccessor) # Include config accessor method
+    end
+
+    # Module providing config accessor method
+    module ConfigAccessor
+      def __async_enumerable_config
+        @async_enumerable_config
+      end
     end
 
     # Async method module - included last to override Enumerable's version.
     module AsyncMethod
```

Comment:
i don't _think_ we need these `AsyncMethod` and `ConfigAccessor` modules -- methods defined in the Enumerable module should automatically get included when `include Async::Enumerable` is called -- the ClassMethods one is a special case since we need to extend it.